core/src/saros/concurrent/management/JupiterClient.java

```diff
@@ -71,4 +68,9 @@ public synchronized ChecksumActivity withTimestamp(ChecksumActivity checksumActi
 
     return get(checksumActivity.getPath()).withTimestamp(checksumActivity);
   }
+
+  // Package-private function for the HeartbeatDispatcher
+  ConcurrentHashMap<SPath, Jupiter> getClientDocs() {
+    return clientDocs;
```

Comment:
I would prefer if you would return an unmodifable view of the map to ensure that the jupiter client still has full control of the map, i.e. no external class can change the content.

If you do adopt this change, this should also be mentioned in the javadoc of the method.
```suggestion
    return return Collections.unmodifiableMap(clientDocs);
```